folding/validators/reward.py

```diff
@@ -1,81 +1,103 @@
-    event["checked_energy"] = [0] * len(uids)
-    event["reported_energy"] = [0] * len(uids)
-    event["miner_energy"] = [0] * len(uids)
-    event["rmsds"] = [0] * len(uids)
-    event["process_md_output_time"] = [0] * len(uids)
-    event["is_run_valid"] = [0] * len(uids)
-
-    energies = np.zeros(len(uids))
-
-    for i, (uid, resp) in enumerate(zip(uids, responses)):
-        # Ensures that the md_outputs from the miners are parsed correctly
-        try:
-            start_time = time.time()
-            can_process = protein.process_md_output(
-                md_output=resp.md_output,
-                hotkey=resp.axon.hotkey,
-                state=resp.miner_state,
-                seed=resp.miner_seed,
-            )
-            event["process_md_output_time"][i] = time.time() - start_time
+class RewardPipeline:
+    def __init__(
+        self, protein: Protein, responses: List[JobSubmissionSynapse], uids: List[int]
+    ):
+        self.protein = protein
+        self.responses = responses
+        self.uids = uids
+
+        self.energies = np.zeros(len(uids))
+
+        self.event = {}
+        self.event["is_valid"] = [False] * len(uids)
+        self.event["checked_energy"] = [0] * len(uids)
+        self.event["reported_energy"] = [0] * len(uids)
+        self.event["miner_energy"] = [0] * len(uids)
+        self.event["rmsds"] = [0] * len(uids)
+        self.event["process_md_output_time"] = [0] * len(uids)
+        self.event["is_run_valid"] = [0] * len(uids)
+
+        self.packages = [None] * len(uids)
+        self.miner_states = [None] * len(uids)
+
+    def process_energies(self):
+        for i, (uid, resp) in enumerate(zip(self.uids, self.responses)):
+            try:
+                start_time = time.time()
+
+                can_process, package = self.protein.process_md_output(
+                    md_output=resp.md_output,
+                    hotkey=resp.axon.hotkey,
+                    state=resp.miner_state,
+                    seed=resp.miner_seed,
+                )
 
-            if not can_process:
-                continue
+                self.packages[i] = package
+                self.miner_states[i] = resp.miner_state
 
-            if resp.dendrite.status_code != 200:
-                bt.logging.info(
-                    f"uid {uid} responded with status code {resp.dendrite.status_code}"
-                )
-                continue
+                self.event["process_md_output_time"][i] = time.time() - start_time
 
-            energy = protein.get_energy()
-            rmsd = protein.get_rmsd()
+                if not can_process:
+                    continue
 
-            if energy == 0:
+                if resp.dendrite.status_code != 200:
+                    bt.logging.info(
+                        f"uid {uid} responded with status code {resp.dendrite.status_code}"
+                    )
+                    continue
+
+                energy = self.protein.get_energy()
```

Comment:
protein.simulation is not updated, will give initial em energy

Reply:
yes, also great point